src/Features/Core/Portable/RemoveAsyncModifier/AbstractRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,236 @@
+
+                    if (controlFlow != null)
+                    {
+                        // For local functions and block bodied lambdas the EndPointIsReachable is false but we still might need to
+                        // insert a return. We can tell by checking for the presence of exit points that aren't return statements
+                        var hasNonReturnExitPoints = controlFlow.ExitPoints.Any<SyntaxNode>(e => !(e is TReturnStatementSyntax));
+                        if (controlFlow.EndPointIsReachable || hasNonReturnExitPoints)
+                        {
+                            var node = GetLastChildOfBlock(replacementNode);
+                            AppendTaskCompletedTaskReturn(node, editor, knownTypes);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void ChangeReturnStatements(SyntaxNode node, SyntaxEditor editor, KnownTypes knownTypes)
+        {
+            var generator = editor.Generator;
+
+            var returns = node.DescendantNodesAndSelf().Where(n => n is TReturnStatementSyntax);
+            foreach (TReturnStatementSyntax returnSyntax in returns)
+            {
+                // Make sure we're not changing returns in nested local functions or lambdas
+                var containingMethod = returnSyntax.FirstAncestorOrSelf<SyntaxNode>(IsAsyncSupportingFunctionSyntax);
+                if (containingMethod != node)
+                {
+                    continue;
+                }
+
+                var returnExpression = generator.SyntaxFacts.GetExpressionOfReturnStatement(returnSyntax);
+                if (returnExpression is null)
+                {
+                    // Convert return; into return Task.CompletedTask;
+                    var returnTaskCompletedTask = GetReturnTaskCompletedTaskStatement(knownTypes, generator);
+                    editor.ReplaceNode(returnSyntax, returnTaskCompletedTask);
+                }
+                else
+                {
+                    // Convert return <expr>; into return Task.FromResult(<expr>);
+                    WrapExpressionWithTaskFromResult(returnExpression, editor, knownTypes);
+                }
+            }
+        }
+
+        private static void AppendTaskCompletedTaskReturn(SyntaxNode lastNode, SyntaxEditor editor, KnownTypes knownTypes)
+        {
+            var generator = editor.Generator;
+            var returnTaskCompletedTask = GetReturnTaskCompletedTaskStatement(knownTypes, generator);
+
+            editor.InsertAfter(lastNode, returnTaskCompletedTask);
+        }
+
+        private static SyntaxNode GetReturnTaskCompletedTaskStatement(KnownTypes knownTypes, SyntaxGenerator generator)
+        {
+            var taskCompletedTaskInvocation = GetTaskCompletedTaskInvocation(knownTypes, generator);
+            var statement = generator.ReturnStatement(taskCompletedTaskInvocation);
+            return statement;
+        }
```

Comment:
nits about all these methods.  totally up to you, but coudl jsut be simple one-line `=>` methods.   they're at that simplicity point for me that that's ok.  but if you prefer the clarity here, that's totally fine.

Reply:
"Negotiating Cyrus' Simplicity Point" will be the title of my memoir 😁